lib/react.dart

```diff
@@ -517,6 +517,20 @@ abstract class Component2 implements Component {
   /// See: <https://reactjs.org/docs/context.html#classcontexttype>
   Context get contextType => null;
 
+  /// Invoked once and cached when [reactComponentClass] is called. Values in the mapping will be set on [props]
+  /// if that prop is not specified by the parent component.
+  ///
+  /// This method is invoked before any instances are created and thus cannot rely on [props]. In addition, be aware
+  /// that any complex objects returned by `defaultProps` will be shared across instances, not copied.
+  ///
+  /// See: <https://reactjs.org/docs/react-component.html#defaultprops>
+  Map get defaultProps => const {};
+
+  /// Invoked once before the `Component` is mounted. The return value will be used as the initial value of [state].
+  ///
+  /// See: <https://facebook.github.io/react/docs/react-component.html#getinitialstate>
```

Comment:
This link is broken. Should link to these sections instead?
- https://reactjs.org/docs/react-without-es6.html#setting-the-initial-state
- https://reactjs.org/docs/react-component.html#constructor

and maybe add a note along the lines of 
> this is similar to getInitialState, and replaces the `this.state =` seen in React ES6 examples
?

Reply:
Or, maybe have an example like the one removed below?
```dart
  /// __Example__:
  ///
  ///     // ES6 component
  ///     constructor() {
  ///       this.state = {count: 0};
  ///     }
  ///
  ///     // Dart component
  ///     @override
  ///     get initialState => {'count': 0};
```